lib/ansible/modules/system/btrfs_quota.py

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2019, Pawel Szatkowski <pszatkowski.byd@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: btrfs_quota
+
+author:
+    - Pawel Szatkowski (@pszatkowski) <pszatkowski.byd@gmail.com>
+
+version_added: '2.8'
+
+short_description: Manage btrfs quota
+
+description:
+    - This module manage btrfs quota
+
+options:
+    state:
+        description:
+            - Enable or disable quota for btrfs filesystem
+        choices: [enabled, disabled]
+        required: true
+    filesystem:
+        description:
+            - Path to btrfs filesystem
+        required: true
+    qgroup:
+        description:
+            - Qgroup ID (0/257) or subvolume path (@/opt)
+        default: none
+        required: false
+    max_rfer:
```

Comment:
```suggestion
    max_reference:
```

Also would be good to further describe below or link to docs.

Reply:
OK, I can put link to btrfs project wiki article that explains what those limits are. 

Regarding to naming suggestion, in my opinion if someone is familiar with btrfs quota, it will know what max_rfer means. Btw this is how this value is being named while listing:

```
root@pawel-VirtualBox:~/btrfs_subvolume# btrfs qgroup show -re /mnt
qgroupid         rfer         excl     max_rfer     max_excl 
--------         ----         ----     --------     -------- 
0/5          16.00KiB     16.00KiB         none         none 
0/257        16.00KiB     16.00KiB         none         none 
0/258        16.00KiB     16.00KiB         none         none 
0/262        16.00KiB     16.00KiB         none         none
```